### PR TITLE
MYST-420 / manager.Connect now sets itself into Connecting state

### PR DESCRIPTION
**Pull request by ignasbernotas**

### CODE REVIEW

**Comment by tadovas** [comment]
What about a test case -> trying to connect, errors occurs - state reverts to not connected ?

File: client/connection/manager.go [hunk 1]
```diff
@@ -43,11 +43,18 @@ func NewManager(mysteriumClient server.Client, dialogEstablisherCreator DialogEs
...
-func (manager *connectionManager) Connect(consumerID, providerID identity.Identity) error {
+func (manager *connectionManager) Connect(consumerID, providerID identity.Identity) (err error) {
 	if manager.status.State != NotConnected {
 		return ErrAlreadyExists
 	}
 
+	manager.status = statusConnecting()
+	defer func() {
+		if err != nil {
+			manager.status = statusNotConnected()
```
Comment (Waldz) [inline client/connection/manager.go]: Would be nice this assert this status in tests

Comment (ignasbernotas) [reply client/connection/manager.go]: It was already covered, but I cleaned up the tests

Comment (ignasbernotas) [reply client/connection/manager.go]: @tadovas @Waldz

File: client/connection/manager.go [hunk 2]
```diff
@@ -43,11 +43,18 @@ func NewManager(mysteriumClient server.Client, dialogEstablisherCreator DialogEs
 	}
 }
 
-func (manager *connectionManager) Connect(consumerID, providerID identity.Identity) error {
+func (manager *connectionManager) Connect(consumerID, providerID identity.Identity) (err error) {
```
Comment (donce) [inline client/connection/manager.go]: It seems this method always returns error explicily (i.e. `return err`). Why do we need this var here?

Comment (ignasbernotas) [reply client/connection/manager.go]: Due to the deferred closure, it needs to know whether there was an error

File: client/connection/manager.go (hunk 1 above)
Comment (Waldz) [reply client/connection/manager.go]: Anyway missing test about status Connecting

Comment (tadovas) [reply client/connection/manager.go]: Added test case. @Waldz  up to you now.

**Pull request merged**